src/bridge/cockpitpipechannel.c

```diff
@@ -384,6 +407,10 @@ cockpit_pipe_channel_prepare (CockpitChannel *channel)
       else
         self->pipe = cockpit_pipe_spawn ((const gchar **)argv, (const gchar **)env, dir, flags);
     }
+  else if (internal && lookup_internal_fd (internal, &fd))
```

Comment:
Well it would have been fine if you had dup()'d it ... then the channel could have been opened more than once. But I think this makes sense, just have it be opened once.